pkg/rancher-desktop/config/__tests__/settings.spec.ts

```diff
@@ -72,10 +249,255 @@ describe('updateFromCommandLine', () => {
+            return jsonProfile;
+          }
+          if (inputPath.endsWith('defaults.plist')) {
+            return plistProfile;
+          }
+          switch (action) {
+          case 1:
+            if (inputPath.endsWith('locked.json')) {
+              return unlockedJSONProfile;
+            } else if (inputPath.endsWith('locked.plist')) {
+              return unlockedPlistProfile;
+            }
+            break;
+          case 2:
+            if (inputPath.endsWith('locked.json')) {
+              return lockedJSONProfile;
+            } else if (inputPath.endsWith('locked.plist')) {
+              return lockedPlistProfile;
+            }
+          }
+          throw new Error("Shouldn't get here.");
+        };
+      }
+
+      describe('when there is no profile', () => {
+        beforeEach(() => {
+          mock = jest.spyOn(fs, 'readFileSync')
+            .mockImplementation(createMocker(0, 0));
+        });
+        afterEach(() => {
+          mock.mockRestore();
+        });
+        test('all fields are unlocked', () => {
+          settings.load();
+          verifyAllFieldsAreUnlocked(settings.getLockedSettings());
+        });
+      });
+
+      describe('when there is only a user profile with unlocked imageList', () => {
+        beforeEach(() => {
+          mock = jest.spyOn(fs, 'readFileSync')
+            .mockImplementation(createMocker(0, 1));
+        });
+        afterEach(() => {
+          mock.mockRestore();
+        });
+        test('all fields are unlocked', () => {
+          settings.load();
+          verifyAllFieldsAreUnlocked(settings.getLockedSettings());
+        });
+      });
+
+      describe('when there is only a user profile with locked imageList', () => {
+        beforeEach(() => {
+          mock = jest.spyOn(fs, 'readFileSync')
+            .mockImplementation(createMocker(0, 2));
+        });
+        afterEach(() => {
+          mock.mockRestore();
+        });
+        test('all fields are locked', () => {
+          settings.load();
+          verifyAllFieldsAreLocked(settings.getLockedSettings());
+        });
+      });
+
+      describe('when there is only a system profile with unlocked imageList', () => {
+        beforeEach(() => {
+          mock = jest.spyOn(fs, 'readFileSync')
+            .mockImplementation(createMocker(1, 0));
+        });
+        afterEach(() => {
+          mock.mockRestore();
+        });
+        test('all fields are unlocked', () => {
+          settings.load();
+          verifyAllFieldsAreUnlocked(settings.getLockedSettings());
+        });
+      });
+
+      describe('when both profiles exit, both with unlocked imageList', () => {
+        beforeEach(() => {
+          mock = jest.spyOn(fs, 'readFileSync')
+            .mockImplementation(createMocker(1, 1));
+        });
+        afterEach(() => {
+          mock.mockRestore();
+        });
+        test('all fields are locked', () => {
```

Comment:
This comment might be backwards.

Reply:
Whatever happened to `cucumber` where the test code is generated from the comments?